packages/frames.js/src/core/middleware/framesjsMiddleware.ts

```diff
@@ -0,0 +1,58 @@
+import { FramesMiddleware, JsonValue } from "../types";
+import { isFrameRedirect, parseButtonInformationFromTargetURL } from "../utils";
+
+type FramesjsMiddlewareContext = {
+  /**
+   * Button that was clicked on previous frame
+   */
+  pressedButton?:
+    | {
+        action: "post" | "post_redirect";
+        index: 1 | 2 | 3 | 4;
+        state?: JsonValue;
+      }
+    | undefined;
+  /** is the initialState for the first frame, and the  */
+  state?: JsonValue | undefined;
```

Comment:
@davidfurlong this is incorrect, the state is not button's state but this middleware uses state from button. `state` should be from `message`.

Reply:
@davidfurlong see `parseFramesMessage` middleware.